### PR TITLE
Update chart's clusters list to be definitive.

### LINKED ISSUE #1942: Support running Kubeapps on a TKG Management Cluster
### Description:

After discussing at length the multi cluster support and adapting it to TKG with @absoludity i believe there is still one key functionality that should be added in the multi cluster area to fully support TKG.

In TKG, the management Cluster is not OIDC enabled, however the Workload Clusters are.
Within the TKG Management Cluster it is suggested to run shared services that provide a unified service toolkit for the workload clusters. this currently is scoped OOTB to include Dex and Harbor (if using TKG+ licensing).

Adding support for running in a non OIDC enabled cluster would be a huge advantage as that would allow kubeapps to follow the TKG methodology and architecture.

### Additional information you deem important (e.g. issue happens only occasionally):

possible solutions i have thought of are:
#### option 1 ####
add 2 additional variables to the helm chart:
1. disableLocalCluster - a boolean that defaults to false and can be set in conjunction with configuring additional Clusters and adding the defaultCluster variable.
2. defaultCluster - the name of the cluster from the additional clusters that should be used as the default cluster when logging in.

#### option 2 ####
add 1 additonal variable to the helm chart:
1. disableLocalCluster - a boolean disabling the local cluster.
in this option there would be no defaultCluster rather there would be a homepage indicating the user must select a cluster. from the context list.

#### Pros and Cons ####
option 1 is probably easier to implement but has 2 main drawbacks:
1. As workload clusters are dynamic and come up and down in certain environments, having a default cluster may prove challenging for some customers.
2. overall in a multi cluster solution you run the risk of customers provisioning charts to the wrong cluster

option 2 is probaly harder to implement but has a few key benefits:
1. makes the user select the cluster which provides an extra security guard around users provisioning charts in the wrong cluster
2. allows kubeapps to be deployed in the management cluster before any additional clusters are deployed and allows them to be more dynamic without making many changes to the kubeapps deployment. (only need to add the scopes and the cluster details in the additional clusters array)

### LINKED ISSUE #1943: Change Default Cluster Name in multi cluster setup
### Description:

Currently in a multi cluster setup, the local cluster which is the default cluster is named default in the context dropdown for selecting clusters. 
As this is a multi cluster solution where i believe all clusters are first class and not just partially supported that it would be better to change this to be the cluster name and not "default".
having it with the name default will confuse users as they wont know what cluster this is.

### PR DESCRIPTION
**Pull request by absoludity**
### Description of the change

Updates the `.Values.clusters` list of clusters to be definitive, enabling the cluster on which kubeapps is installed to have a name other than `default` or to not be a target cluster at all for users.

### Benefits

See #1942 and #1943 

TL;DR is that with this change, the cluster on which Kubeapps is running can be called whatever the user likes, but still defaults to `default`, and this is included in the definitive list of `.Values.clusters`.

### Applicable issues

  - fixes #1942 
  - fixes #1943

### CODE REVIEW

**Comment by absoludity** [comment]
Marking this ready for review, it still has one CI test failing which I need to look into (fail when missing permissions), but I'll do that first thing.

File: dashboard/src/reducers/cluster.ts [hunk 1]
```diff
@@ -146,20 +146,19 @@ const clusterReducer = (
...
       break;
     case getType(actions.config.receiveConfig):
-      // Initialize the additional clusters when receiving the config.
-      const clusters = {
-        ...state.clusters,
-      };
+      // Initialize the clusters when receiving the config.
       const config = action.payload as IConfig;
+      const clusters: IClustersMap = {};
       config.clusters?.forEach(cluster => {
```
Comment (andresmgot) [inline dashboard/src/reducers/cluster.ts:152]: so `clusters` should be defined, right? `config.clusters.forEach`

**Review by andresmgot — APPROVED** [review]

File: dashboard/src/reducers/cluster.ts (hunk 1 above)
Comment (absoludity) [reply dashboard/src/reducers/cluster.ts:152]: Yep, indeed.
